dna/modify.go

```diff
@@ -133,6 +139,23 @@ func Insert(seq []Base, insPos int64, insSeq []Base) []Base {
 	return append(seq[:insPos], append(insSeq, seq[insPos:]...)...)
 }
 
+// InsertStable is identical to Insert, but conserves the
+// underlying memory of the input destSeq slice if possible
+func InsertStable(destSeq []Base, insPos int, insSeq []Base) ([]Base, error) {
```

Comment:
I could also go either way. I definitely see the benefit (and prefer the style) of not returning a `[]Base`. Just to lay out my logic, the reason I built it to have a return was so that if the insertion exceeded the dest slice capacity then it would automatically generate a new slice and return an error. Do you think this tries to predict the users needs too much? I could just return the error and leave it to the user to generate a copy if they want. Interested in hearing your opinion.

Reply:
I think your idea is a good one to allocate new memory for the user (if needed) and then use this new slice if needed.  I think you can do this seamlessly by passing in a pointer to the slice instead of a slice.  The trick is that at the end of the function when you would normally return the slice, you instead set the memory pointed to by the pointer to the values that reference the new slice.  This usually takes the form of:
*in = answer
I made a toy example you can see here:
https://play.golang.org/p/negQb7k_jkd